pkg/provider/configuration.go

```diff
@@ -77,6 +81,13 @@ func Merge(ctx context.Context, configurations map[string]*dynamic.Configuration
 			}
 		}
 
+		for transportName, transport := range conf.HTTP.ServersTransports {
+			transports[transportName] = append(transports[transportName], root)
+			if !AddTransport(configuration.HTTP, transportName, transport) {
+				transportsToDelete[transportName] = struct{}{}
+			}
+		}
```

Comment:
Afaiu, this should have already been added before this PR, for the Merge to work with any provider that already actually were dealing with a ServersTransport. So I don't get why it was missing until now, and how that was not causing any problem.
Do you know?

Reply:
Not really sure, that code was written by @Gufran; I'll try to see if I can find something over the weekend. Do you know another provider utilizing this, maybe I can dig there…

Reply:
Ah the kubernetes CRD provider creates the config without merging; so I think that missing change might simply be an oversight in traefik for now?

Reply:
yes, that is the only explanation I found for now. But I find a bit surprising that nobody complained or that it does not break anything in any obvious way. ServersTransport is also supposed to work in the file provider btw.

Reply:
uh, as far as I can see, neither the file provider nor the kubernetes CRD one explicitly call provider.Merge (whereas most of the others do), which might explain why this omission went unnoticed so far.

Reply:
Also, I don't remember what that does, but I notice the `ServersTransports` in `HTTPConfiguration` has `label:"-"`, and I now wonder if it should.

Reply:
Me neither, I am rather new to go and am not exactly sure how to find out what actually acts on those tags :/

Reply:
yeah, we should keep it like that for now, in order to prevent ServersTransports being configurable through labels.